### PR TITLE
Add setcmdline()

### PR DESCRIPTION
**Pull request by Shougo**
It adds `setcmdline()` function.

### CODE REVIEW

**Comment by Shougo** [comment]
I will port it to Vim.

**Comment by Shougo** [comment]
`pum.vim` supports the feature.

File: src/nvim/eval/funcs.c [hunk 1]
```diff
@@ -8168,6 +8168,17 @@ static void f_setcharsearch(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   }
 }
 
+/// "setcmdline()" function
+static void f_setcmdline(typval_T *argvars, typval_T *rettv, FunPtr fptr)
```
Comment (bfredl) [inline src/nvim/eval/funcs.c:8172]: let's take cmdpos as an optional second parameter (`setcmdpos()` will not work, it needs to be changed directly in `set_cmdline_str`)

File: src/nvim/ex_getln.c [hunk 2]
```diff
@@ -6250,6 +6250,58 @@ int get_cmdline_screen_pos(void)
...
+    restore_v_event(dict, &save_v_event);
+
+    bool tl_ret = try_leave(&tstate, &err);
+    if (!tl_ret && ERROR_SET(&err)) {
+      msg_putchar('\n');
+      msg_printf_attr(HL_ATTR(HLF_E)|MSG_HIST, (char *)e_autocmd_err, err.msg);
+      api_clear_error(&err);
+      redrawcmd();
+    }
+  }
```
Comment (bfredl) [inline src/nvim/ex_getln.c:6300]: this is quite a bit of duplicated code, could break out to a `trigger_cmdlinechanged_event` helper function

File: src/nvim/eval/funcs.c (hunk 1 above)
Comment (Shougo) [reply src/nvim/eval/funcs.c:8172]: Fixed.

**Comment by Shougo** [comment]
I have ported it to Vim.

https://github.com/vim/vim/pull/10869

**Comment by Shougo** [comment]
It is merged in Vim.

https://github.com/neovim/neovim/pull/19979

**Pull request closed without merging**